src/Collector.cs

```diff
@@ -153,5 +143,26 @@ private static Socket Connect()
       _log.Info("Connected to collector.");
       return socket;
     }
+
+    private static bool IsSocketDisconnected (Socket socket)
+    {
+      // Is there any data available?
+      byte[] buffer = null;
+      while (socket.Poll(1, SelectMode.SelectRead))
+        {
+          // If no data is available then socket disconnected
+          if (socket.Available == 0)
+            return true;
```

Comment:
I must be missing something, but it looks like this will report the connection as disconnected whenever there is no data to be read. Given the collectors send only 2 initial messages, it seems like this would cause regular, predictable false-positives for disconnection.